plugin/plugin_test.go

```diff
@@ -4,8 +4,52 @@
 
 package plugin
 
-import "testing"
+import (
+	"testing"
 
-func TestPlugin(t *testing.T) {
-	t.Skip()
+	"github.com/stretchr/testify/assert"
+)
+
+func TestValidateArgs(t *testing.T) {
+	keyExists := Args{ApiKey: "someKey"}
+	noKey := Args{ApiKey: ""}
+
+	// assert equality
+	assert.Nil(t, validateAndSetArgs(&keyExists), "result is nil")
+	assert.NotNil(t, noKey, "result is an error")
```

Comment:
can you provide some more context regarding this line here?

Reply:
If there was an error it would return it, if not nil. my understanding of the above line is, it expects an error therefore it shouldn't be nil? I've removed the "result is an error" part. Not required.